### PR TITLE
Add support for FFmpeg-Vaapi drawbox filter

### PR DESCRIPTION
**Pull request by xhaihao**

### CODE REVIEW

File: lib/ffmpeg/vaapi/vpp.py [hunk 1]
```diff
@@ -42,6 +42,25 @@ def gen_vpp_opts(self):
...
+      if hasattr(self, "x"):
+        vpopts.append(f"x={self.x}")
+      if hasattr(self, "y"):
+        vpopts.append(f"y={self.y}")
+      if hasattr(self, "color"):
+        vpopts.append(f"color={self.color}")
+      if hasattr(self, "t"):
+        vpopts.append(f"t={self.t}")
+      if hasattr(self, "replace"):
+        vpopts.append(f"t={self.replace}")
```
Comment (uartie) [inline lib/ffmpeg/vaapi/vpp.py:61]: Both use the same ffmpeg option `t`?

File: lib/mixin/vpp.py [hunk 2]
```diff
@@ -224,3 +224,15 @@ def check_color_props(self):
     metrics2.check(**params)
 
   check_range  = check_color_props
+
+  def check_drawbox(self):
+    owidth, oheight = self.width, self.height
```
Comment (uartie) [inline lib/mixin/vpp.py:229]: Assignment not needed.  Just pass directly to the `filesize` check

File: lib/mixin/vpp.py [hunk 3]
```diff
@@ -224,3 +224,15 @@ def check_color_props(self):
...
+
+    metrics2.check(
+      metric = dict(type = "filesize"),
+      filetest = self.decoded, width = owidth, height = oheight,
+      frames = self.frames, format = self.format)
+
+    params = vars(self).copy()
+    params.update(width = owidth, height = oheight)
+    metrics2.check(**params)
```
Comment (uartie) [inline lib/mixin/vpp.py:238]: Copy/update is unnecessary... nothing changed.  How about call `self.check_default()` instead?

**Review by uartie — CHANGES_REQUESTED** [review]

File: lib/ffmpeg/vaapi/vpp.py (hunk 1 above)
Comment (xhaihao) [reply lib/ffmpeg/vaapi/vpp.py]: Thanks for catching this typo, fixed in the new version.

File: lib/mixin/vpp.py (hunk 2 above)
Comment (xhaihao) [reply lib/mixin/vpp.py]: Thanks for pointing it out, updated.

File: lib/mixin/vpp.py (hunk 3 above)
Comment (xhaihao) [reply lib/mixin/vpp.py]: Updated, thx

**Review by uartie — APPROVED** [review]

**Pull request merged**